Frontend/src/auth/context/firebase/auth-provider.tsx

```diff
@@ -0,0 +1,214 @@
+import { useMemo, useEffect, useReducer, useCallback } from 'react';
+import {
+  getAuth,
+  signOut,
+  OAuthProvider,
+  signInWithPopup,
+  onAuthStateChanged,
+  GoogleAuthProvider,
+  FacebookAuthProvider,
+} from 'firebase/auth';
+
+// import axios, { endpoints } from 'src/utils/axios';
+
+import { AuthContext } from './auth-context';
+import { firebaseApp, isRegistered as isRegister } from './lib';
+import { AuthUserType, ActionMapType, AuthStateType } from '../../types';
+// ----------------------------------------------------------------------
+
+const AUTH = getAuth(firebaseApp);
+
+enum Types {
+  INITIAL = 'INITIAL',
```

Comment:
what you mean by INITIAL and REGISTER? can you explain. If you add a comment in the code that would be better.

I am expecting one to be for firebase and another one to be for full authentication with our app. if that is the case you can name this as such and you won't need comment:
`INITIAL_FIREBASE_AUTH`
`FULL_APP_AUTH` I think will be more readable